wandb/sdk/wandb_settings.py

```diff
@@ -171,11 +171,11 @@ def _get_program() -> Optional[str]:
 
 
 def _get_program_relpath_from_gitrepo(
-    program: str, _logger: Optional[_EarlyLogger] = None
+    program: str, disable_git: Optional[bool], _logger: Optional[_EarlyLogger] = None
 ) -> Optional[str]:
     repo = GitRepo()
     root = repo.root
-    if not root:
+    if not root or disable_git:
```

Comment:
this is the crux of the fix, when git is disabled we **dont** want to rely on root, this breaks entrypoints that start in the middle of a subdir structure in nested git repos.